File: src/main/java/com/exmple/kyudo_app/domain/service/UserRegisterService.java
package com.exmple.kyudo_app.domain.service;

import com.exmple.kyudo_app.domain.model.User;
import com.exmple.kyudo_app.infrastructure.external.UserRegisterRepository;
import org.springframework.beans.factory.annotation.Autowired;
import org.springframework.stereotype.Service;
import org.springframework.transaction.annotation.Transactional;

@Service
public class UserRegisterService {

    private final UserRegisterRepository userRegisterRepository;

    @Autowired
    public UserRegisterService(UserRegisterRepository userRegisterRepository){
        this.userRegisterRepository = userRegisterRepository;
    }

@Transactional
    public User registryUser(User user){
        return userRegisterRepository.save(user);
}

@Transactional
    public User getUserById(Long id){
        return userRegisterRepository.getReferenceById(id);
}
}

File: src/main/kotlin/com/example/kyudo_app/domain/service/UserRegisterService.kt
package com.example.kyudo_app.domain.service

import com.example.kyudo_app.domain.model.User
import com.example.kyudo_app.infrastructure.external.UserRegisterRepository
import org.springframework.stereotype.Service
import org.springframework.transaction.annotation.Transactional
import org.springframework.data.repository.findByIdOrNull

@Service
class UserRegisterService(
    private val userRegisterRepository: UserRegisterRepository
) {
    @Transactional
    fun registryUser(user: User): User {
        return userRegisterRepository.save(user)
    }

    @Transactional(readOnly = true)
    fun getUserById(id: Long): User {
        return userRegisterRepository.findByIdOrNull(id)
            ?: throw IllegalArgumentException("User not found for id=$id")
    }
}
